sessionctx/context.go

```diff
@@ -149,3 +152,40 @@ const (
 	// LastExecuteDDL is the key for whether the session execute a ddl command last time.
 	LastExecuteDDL basicCtxType = 3
 )
+
+// ValidateSnapshotReadTS strictly validates that readTS does not exceed the PD timestamp
+func ValidateSnapshotReadTS(ctx context.Context, sctx Context, readTS uint64) error {
+	txnScope := sctx.GetSessionVars().CheckAndGetTxnScope()
```

Comment:
Maybe we should use `oracle.GlobalTxnScope` directly?

Reply:
Ah, yes. This looks safer.